cherrypy/lib/sessions.py

```diff
@@ -566,8 +565,6 @@ def acquire_lock(self, path=None):
     def release_lock(self, path=None):
         """Release the lock on the currently-loaded session data."""
         self.lock.close()
-        with contextlib.suppress(FileNotFoundError):
-            os.remove(self.lock._path)
```

Comment:
I think we need something smarter than not removing files.

Reply:
My instinct is that less code is better (especially since the file lock should ideally be managed by zc.lockfile). It looks like this change was proposed way back in #1391 also. I don't like the idea of leaving artifacts around after exiting, but since Unix allows two manifestations of the same file to exist at the same time, I don't think there's any option but to always leave the file.

Reply:
I'm curious whether we could somehow introduce better locking here.

Reply:
In my experience, locking is hard to get right, which is why I've been seeking to rely on libraries that are battle tested and relied upon by other frameworks. I'd be open to another implementation, but would prefer if CherryPy didn't have to host it. I don't see anything that CherryPy is doing special that warrants a custom lock behavior.

Reply:
I absolutely agree that it should be fully implemented in a separate dist.